components/switch/src/styled.js

```diff
@@ -0,0 +1,147 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { SrOnly } from '../../accessibility-helpers/src';
+
+// ==============================
+// Utils
+// ==============================
+
+const asArray = val => (Array.isArray(val) ? val : [val]);
+
+// ==============================
+// Component
+// ==============================
+
+export const SwitchText = ({ size, block, flip, srOnlyText, ...props }) => {
+	const { breakpoints, switch: formSwitch } = useTheme();
+	const mq = paint(breakpoints);
+
+	const styleText = {
+		flex: block ? 1 : null,
+		display: 'flex',
+		alignItems: 'center',
+		minHeight: asArray(size).map(s => s && formSwitch.size[s].height),
+		...(gap => (flip ? { paddingLeft: gap } : { paddingRight: gap }))(formSwitch.text.gap),
+
+		'input:disabled ~ &': {
+			cursor: 'default',
+			...formSwitch.toggleText.disabled,
+		},
+	};
+
+	return srOnlyText ? <SrOnly {...props} /> : <span css={mq(styleText)} {...props} />;
+};
+
+export const SwitchToggle = ({ size, toggleText, ...props }) => {
+	const { breakpoints, switch: formSwitch } = useTheme();
+	const mq = paint(breakpoints);
+
+	const styleResponsive = () => {
+		const sizeArr = asArray(size);
+
+		return {
+			borderRadius: sizeArr.map(s => s && formSwitch.size[s].borderRadius),
+			width: sizeArr.map(s => s && formSwitch.size[s].width),
+			height: sizeArr.map(s => s && formSwitch.size[s].height),
+			fontSize: sizeArr.map(s => s && formSwitch.size[s].fontSize),
+		};
+	};
+
+	// On/off toggle switch styling
+	const styleToggle = {
+		borderRadius: styleResponsive().borderRadius,
+		width: styleResponsive().width,
+		height: styleResponsive().height,
```

Comment:
For optimisation call the styleReponsive function once and assign to a variable. i.e. 
```
const responsive = styleResponsive();

borderRadius: responsive.borderRadius
```
(feel free to use whatever name)